neuro_san/internals/run_context/langchain/journaling/journaling_callback_handler.py

```diff
@@ -137,35 +138,29 @@ async def on_tool_start(
             passed to the tool.
         """
 
-        if "langchain_tool" in tags:
+        # Extract tool name from the serialized data
+        agent_name: str = serialized.get("name")
 
-            # Extract tool name from the serialized data
-            agent_name: str = serialized.get("name")
+        # Remove any policy objects from the arguments passed in.
+        tool_start_dict: Dict[str, Any] = ToolArgumentReporting.prepare_tool_start_dict(inputs)
 
-            # Build the origin path
-            self.origin: List[Dict[str, Any]] = self.origination.add_spec_name_to_origin(self.parent_origin, agent_name)
-            full_name: str = self.origination.get_full_name_from_origin(self.origin)
+        # Report that we are about to invoke a tool.
+        content: str = f"Invoking: `{agent_name}` with `{tool_start_dict.get('tool_args')}`"
+        message: BaseMessage = AgentMessage(content=content)
+        await self.calling_agent_journal.write_message(message)
 
-            # Combine the original tool inputs with origin metadata
-            combined_args: Dict[str, Any] = inputs.copy()
-            combined_args["origin"] = self.origin
-            combined_args["origin_str"] = full_name
```

Comment:
This comibined_args business was common logic that was factored out into ToolArgumentReporting.